library/modules/Buildings.cpp

```diff
@@ -670,10 +676,17 @@ std::pair<df::coord,df::coord2d> Buildings::getSize(df::building *bld)
     return std::pair<df::coord,df::coord2d>(pos, df::coord2d(bld->x2+1,bld->y2+1) - pos);
 }
 
-static bool checkBuildingTiles(df::building *bld, bool can_change)
+static bool checkBuildingTiles(df::building *bld, bool can_change,
+                               bool force_extents = false)
 {
     auto psize = Buildings::getSize(bld);
 
+    if (force_extents && !bld->room.extents)
```

Comment:
Given that only one caller sets `force_extents` currently, would it make sense to split this bit into a separate function? Seems a bit out of place in a `check` function to me. Granted, it is internal-only, and it looks like `checkFreeTiles()` can (or could?) modify buildings as well.

Reply:
I don't quite follow your meaning here. Are you suggesting moving this `if` statement to `constructAbstract()` or making it independently callable external to the `Buildings` module?

Some context that may or may not be relevant to your statement: the initialization code was already in `checkFreeTiles()`, so I'm not adding anything new there (though I agree that modification of the structure is not implied by the "check" name), I just refactored the existing code into a separate function so it's callable from more places. This `if` statement is here instead of in `constructAbstract` since we need to set the building size before allocating the extent.

Reply:
I guess I was advocating for moving the logic to a separate function (still internal) with a better name, but with context, it sounds like that isn't important. (Basically, it looked to me like a "checkX()" function was changing to "checkXAndMaybeDoY()", with only one caller relying on behavior Y, but I didn't realize you were just moving the Y logic around.)